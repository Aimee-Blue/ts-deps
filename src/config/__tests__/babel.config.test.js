const { default: config, babelConfig } = require('../babel.config')
const { validate } = require('@babel/core/lib/config/validation/options')

describe('given no options', () => {
  it('should work', () => {
    const defaultConfig = babelConfig()
    expect(defaultConfig).toMatchSnapshot()
    expect(defaultConfig).toMatchObject(config)

    expect(() => validate('configfile', defaultConfig)).not.toThrow()
  })
})

describe('given node version', () => {
  it('should work', () => {
    const cfg = babelConfig({
      nodeVersion: '10',
    })
    expect(cfg).toMatchSnapshot()

    expect(() => validate('configfile', cfg)).not.toThrow()
  })
})

describe('excluding tests in the output', () => {
  it('should work', () => {
    const cfg = babelConfig({
      ignoreTests: true,
    })
    expect(cfg).toMatchSnapshot()

    expect(() => validate('configfile', cfg)).not.toThrow()
  })
})

describe('given aliases', () => {
  it('should work', () => {
    const cfg = babelConfig({
      aliases: {
        ['@shared']: '/src/shared',
        ['@feature']: 'src/feature/',
      },
    })
    expect(cfg).toMatchSnapshot()

    expect(() => validate('configfile', cfg)).not.toThrow()
  })
})

describe('given options', () => {
  it('should work', () => {
    const cfg = babelConfig({
      aliases: {
        ['@shared']: '/src/shared/',
      },
      nodeVersion: '10',
    })
    expect(cfg).toMatchSnapshot()

    expect(() => validate('configfile', cfg)).not.toThrow()
  })
})
