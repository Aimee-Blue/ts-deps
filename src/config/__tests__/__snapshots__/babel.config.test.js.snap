// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`excluding tests in the output should work 1`] = `
Object {
  "ignore": Array [
    "src/**/*.d.ts",
    "src/**/*.json",
  ],
  "plugins": Array [
    Array [
      "module-resolver",
      Object {
        "alias": Object {
          "ts-deps": "./src",
        },
        "root": Array [
          "src",
        ],
      },
    ],
    "@babel/proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-optional-chaining",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "targets": Object {
          "node": "8.12",
        },
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`given aliases should work 1`] = `
Object {
  "ignore": Array [
    "src/**/*.d.ts",
    "src/**/*.json",
  ],
  "plugins": Array [
    Array [
      "module-resolver",
      Object {
        "alias": Object {
          "@feature": "src/feature",
          "@shared": "src/shared",
        },
        "root": Array [
          "src",
        ],
      },
    ],
    "@babel/proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-optional-chaining",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "targets": Object {
          "node": "8.12",
        },
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`given no options should work 1`] = `
Object {
  "ignore": Array [
    "src/**/*.d.ts",
    "src/**/*.json",
  ],
  "plugins": Array [
    Array [
      "module-resolver",
      Object {
        "alias": Object {
          "ts-deps": "./src",
        },
        "root": Array [
          "src",
        ],
      },
    ],
    "@babel/proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-optional-chaining",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "targets": Object {
          "node": "8.12",
        },
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`given node version should work 1`] = `
Object {
  "ignore": Array [
    "src/**/*.d.ts",
    "src/**/*.json",
  ],
  "plugins": Array [
    Array [
      "module-resolver",
      Object {
        "alias": Object {
          "ts-deps": "./src",
        },
        "root": Array [
          "src",
        ],
      },
    ],
    "@babel/proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-optional-chaining",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "targets": Object {
          "node": "10",
        },
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`given options should work 1`] = `
Object {
  "ignore": Array [
    "src/**/*.d.ts",
    "src/**/*.json",
  ],
  "plugins": Array [
    Array [
      "module-resolver",
      Object {
        "alias": Object {
          "@shared": "src/shared",
        },
        "root": Array [
          "src",
        ],
      },
    ],
    "@babel/proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-optional-chaining",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "targets": Object {
          "node": "10",
        },
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;
